diag-control.h

```diff
@@ -37,9 +37,13 @@
 
 #include "pcap/compiler-tests.h"
 
-#if PCAP_IS_AT_LEAST_CLANG_VERSION(2,8) || PCAP_IS_AT_LEAST_GNUC_VERSION(4,6)
+#if PCAP_IS_AT_LEAST_CLANG_VERSION(2,8) || \
+    PCAP_IS_AT_LEAST_GNUC_VERSION(4,6) || \
+    PCAP_IS_AT_LEAST_SUNC_VERSION(5,5)
   /*
    * Clang and GCC both support this way of putting pragmas into #defines.
+   * Sun C compiler version 5.5 (Studio version 8) and later supports both
```

Comment:
You can probably say "Clang 2.8 and later, GCC 4.6 and later, and Sun C 5.5 (Studio version 8) all support this way of putting pragmas into #defines", as that's all that part of the header does.

All three of them require different arguments to `PCAP_DO_PRAGMA()`, so that's handled elsewhere.

Reply:
Hopefully in the latest revision the comments are good enough.